process_report/processors/coldfront_fetch_processor.py

```diff
@@ -0,0 +1,104 @@
+import os
+import functools
+import logging
+from dataclasses import dataclass
+
+import requests
+
+from process_report.invoices import invoice
+from process_report.processors import processor
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+CF_ATTR_ALLOCATED_PROJECT_NAME = "Allocated Project Name"
+
+
+@dataclass
+class ColdfrontFetchProcessor(processor.Processor):
+    nonbillable_projects: list[str]
+
+    @functools.cached_property
+    def coldfront_client(self):
+        keycloak_url = os.environ.get("KEYCLOAK_URL", "https://keycloak.mss.mghpcc.org")
+
+        # Authenticate with Keycloak
+        token_url = f"{keycloak_url}/auth/realms/mss/protocol/openid-connect/token"
+        try:
+            r = requests.post(
+                token_url,
+                data={"grant_type": "client_credentials"},
+                auth=requests.auth.HTTPBasicAuth(
+                    os.environ["KEYCLOAK_CLIENT_ID"],
+                    os.environ["KEYCLOAK_CLIENT_SECRET"],
+                ),
+            )
```

Comment:
You never check for HTTP errors. One option is to call `r.raise_for_status()` inside the `try/except` block, and then add an `except` stanza that will translate http errors into a more user friendly error message.

Reply:
@QuanMPhm ^^ Please check for errors and log a useful error message.

Somebody accidentally changed my secret (which sets the KEYCLOAK_CLIENT* env variables) and I was wondering why my openshift report was failing. If it logged the http message (in my case 401 for unauthorized) and the client_id I would have immediately spotted the problem instead of debugging the code in the container because it worked fine on my local machine.